src/main/java/uk/gov/hmcts/reform/authorisation/ServiceDetailClient.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.authorisation;
+
+import feign.FeignException;
+import uk.gov.hmcts.reform.authorisation.exceptions.AbstractAuthorisationException;
+
+public class ServiceDetailClient {
```

Comment:
perhaps we could just add an extra method to `ServiceAuthTokenValidator`?

Reply:
i was afraid if i do, some one would have commented "it does not represent any validation" or something like that

on the other hand, client can incorporate multiple interfaces if we want to and really just use client in a project but specific interface in relevant place. just a thought

Reply:
it actually does and returns service name 'as a bonus'

Reply:
it does, but does not represent as in endpoint definition: just looking at `GET /details` does imply validation

Reply:
poor route naming in s2s doesn't have to have an impact on functions in this java library imho :)

Reply:
_fairs enoughs_.. i'll get rid of extra class